### PR TITLE
[MODULAR] Adds a new martial art to uplinks, Tribal Claw.

### PR DESCRIPTION
**Pull request by Zonespace27**
## About The Pull Request
Ports (with some edits) BeeStation/BeeStation-Hornet#3011 and associated fix PRs
Closed by upstream in part due to me not properly understanding the species change policy, ontop of Oranges not liking it

This PR adds a new martial art to traitor uplinks for lizards and Unathi for 13 TC.
The martial art of tribal claw is a lizard-exclusive marital art that lets lizards utilize their tail and claws to do extreme damage to their opponents with combos.

## Moveset
**Disarm, Disarm, Grab, Harm**: Tail Sweep
Tail sweep is a combo move that lets a lizard spin around, sending anyone within 1 tile of them flying 4 tiles away. Mechanically, this move operates very similarly to the xeno tail spin.
**Harm, Disarm**: Face Scratch
Face scratch is a very simple combo move that deals an extra 10 (non-armor piercing) brute damage, on top of 5 confusion and 5 eye blur (spraying someone with a spray can applies 3, for reference).
**Harm, Harm, Grab**: Jugular Cut
Jugular cut is one of the stronger moves that Tribal Claw grants, dealing 15 (non-armor piercing) brute damage to the head, on top of applying a severe wound to the head and the "slit throat" status effect. The pre-requisite to being able to use this ability is that the target either needs to: be in critical condition, be in a neck choke by the attacker, or be asleep.
**Disarm, Harm, Grab, Grab**: Tail Grab
Intended to combo with jugular cut, the tail grab is a strong move at allows the attacker to disable the defender with ease. On executing this move, the defender will be instantly put in a chokehold by the attacker, in addition to being silenced for a short period of time.

## How this compares to other martial arts
https://imgur.com/a/fmKAwf9
Imgur gallery has videos and the ideal TTK of each martial art, Tribal Claw is on the mediocre side for straight up killing someone with the martial art alone.

## Why It's Good For The Game
Martial arts are good and fun, and I saw this one elsewhere and thought it was neat. In addition, there's not much in the way of race-restricted items (a whole **1**), and this helps increase the amount by double.

## Changelog

:cl:
add: Added tribal claw martial art to lizard and unathi traitor uplinks for 13 TC
/:cl:

### CODE REVIEW

**Comment by RatFromTheJungle** [comment]
i'd mimic upstream's reasoning of not wanting species locked martial arts, feels to close to 'made for me', i'd also question the validity of the balance around the tail-sweep and jugular grab, considering the only other martial arts with strong entry combos like that is CQC, for almost double the price this is set at.

**Comment by Zonespace27** [comment]
> I'd also question the validity of the balance around the tail-sweep and jugular grab, considering the only other martial arts with strong entry combos like that is CQC, for almost double the price this is set at.

CQC is capable of KOing someone in 3 hits (two grabs and a disarm), this requires 4 hits for a tier 3 grab instead. In addition, Sleeping Carp (which this is currently on-par with price-wise), is able of knocking someone to the ground in a fairly braindead 2 hit combo (Punch grab), and repeating the same 2 hit combo will paralyze someone. Not to mention that one of scarp's strong points is the bullet reflection, and CQC's is the massive element of surprise that it grants (instant aggressive grabs, KO in 3 hits, knockdown in 2), this martial art lacks both, except for the short time that you're silenced when you're tail grabbed, which i'm not *too* attached to.

**Comment by RatFromTheJungle** [comment]
> > I'd also question the validity of the balance around the tail-sweep and jugular grab, considering the only other martial arts with strong entry combos like that is CQC, for almost double the price this is set at.
> 
> CQC is capable of KOing someone in 3 hits (two grabs and a disarm), this requires 4 hits for a tier 3 grab instead. In addition, Sleeping Carp (which this is currently on-par with price-wise), is able of knocking someone to the ground in a fairly braindead 2 hit combo (Punch grab), and repeating the same 2 hit combo will paralyze someone. Not to mention that one of scarp's strong points is the bullet reflection, and CQC's is the massive element of surprise that it grants (instant aggressive grabs, KO in 3 hits, knockdown in 2), this martial art lacks both, except for the short time that you're silenced when you're tail grabbed, which i'm not _too_ attached to.



> > I'd also question the validity of the balance around the tail-sweep and jugular grab, considering the only other martial arts with strong entry combos like that is CQC, for almost double the price this is set at.
> 
> CQC is capable of KOing someone in 3 hits (two grabs and a disarm), this requires 4 hits for a tier 3 grab instead. In addition, Sleeping Carp (which this is currently on-par with price-wise), is able of knocking someone to the ground in a fairly braindead 2 hit combo (Punch grab), and repeating the same 2 hit combo will paralyze someone. Not to mention that one of scarp's strong points is the bullet reflection, and CQC's is the massive element of surprise that it grants (instant aggressive grabs, KO in 3 hits, knockdown in 2), this martial art lacks both, except for the short time that you're silenced when you're tail grabbed, which i'm not _too_ attached to.

i hate to be the one to tell you of this, but the repulse move you gave has an inbuilt hardstun, I believe 7s from repulse itself (which if I read it correctly, is a 1 tile range around the user from the martial-art).
and this also doesn't come with any of the downsides of those two martial arts, which is very important to mention.
(cqc replaces all your grabs, making it painfully obvious when you have it, and carp blocks you from using guns)

**Comment by Zonespace27** [comment]
> but the repulse move you gave has an inbuilt hardstun, I believe 7s from repulse itself.

Yeah that can afford to go, from my understanding it should simply be a "get away from me" ability, not a fight-ender.

**Comment by RatFromTheJungle** [comment]
a 7s hardstun is enough to cuff, or just kick into stamcrit at at whichpoint the fight is over by that combo in specific.

**Comment by Zonespace27** [comment]
Lowered hardstun time to 1.5 seconds from 6, lowered floor slam hardstun (when you do tail sweep on the same tile as the defender) to 4 seconds from 10

**Comment by YakumoChen** [comment]
CQC was meant to be locked to nukeops and is intentionally insanely powerful, I'm not sure on having a half priced cqc with a disable and an execute.
As said before, this also has no real downside to using, no "tell" like the replacing gun use or grabs.

**Comment by Zonespace27** [comment]
> CQC was meant to be locked to nukeops and is intentionally insanely powerful, I'm not sure on having a half priced cqc with a disable and an execute. As said before, this also has no real downside to using, no "tell" like the replacing gun use or grabs.

I'll playtest it a bit more, and possibly look into either:

- Raising the price
- Giving it some form of tell
- making the combos more complex in some way

However, keep in mind that it has a much higher raw time to kill than CQC, SCarp, or plasma fist, as shown in [this](https://imgur.com/a/fmKAwf9) imgur album that i put in the PR body

**Comment by Iamgoofball** [comment]
please ensure all martial arts are priced equally to sleeping carp in both TC cost and progression minimums

**Comment by Zonespace27** [comment]
> please ensure all martial arts are priced equally to sleeping carp in both TC cost and progression minimums

Currently equal price to sleeping carp, forgot to add the progression time, made that equal as well

File: modular_skyrat/master_files/code/datums/martial/tribal_claw.dm [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	return FALSE
+
+//Tail Sweep, triggers an effect similar to a xeno's tail sweep but only affects stuff 1 tile next to you, basically 3x3.
+/datum/martial_art/tribal_claw/proc/tailSweep(mob/living/carbon/attacker, mob/living/carbon/defender)
+	if(attacker == current_target)
+		return
+	log_combat(attacker, defender, "tail sweeped(Tribal Claw)")
+	defender.visible_message("<span class='warning'>[attacker] sweeps [defender]'s legs with their tail!</span>", \
+						"<span class='userdanger'>[attacker] sweeps your legs with their tail!</span>")
+	var/obj/effect/proc_holder/spell/aoe_turf/repulse/lizard_tail/tail_hit = new
```
Comment (Fikou) [inline modular_skyrat/master_files/code/datums/martial/tribal_claw.dm:39]: lol

File: modular_skyrat/master_files/code/modules/spells/tribal_claw.dm [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	action_background_icon_state = "bg_alien"
+	anti_magic_check = FALSE
+	maxthrow = 4
+	floor_slam_stun = 4 SECONDS
+
+/obj/effect/proc_holder/spell/aoe_turf/repulse/lizard_tail/cast(list/targets, mob/user = usr)
+	if(iscarbon(user))
+		var/mob/living/carbon/C = user
+		playsound(C.loc,'sound/effects/hit_punch.ogg', 80, 1, 1)
+		C.spin(6,1)
```
Comment (Fikou) [inline modular_skyrat/master_files/code/modules/spells/tribal_claw.dm]: use named args if possible

File: modular_skyrat/master_files/code/modules/spells/tribal_claw.dm [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	action_icon = 'icons/mob/actions/actions_xeno.dmi' // I lack anything better for this, but you shouldn't be seeing this anyway
+	action_icon_state = "tailsweep"
+	action_background_icon_state = "bg_alien"
+	anti_magic_check = FALSE
+	maxthrow = 4
+	floor_slam_stun = 4 SECONDS
+
+/obj/effect/proc_holder/spell/aoe_turf/repulse/lizard_tail/cast(list/targets, mob/user = usr)
+	if(iscarbon(user))
+		var/mob/living/carbon/C = user
```
Comment (Fikou) [inline modular_skyrat/master_files/code/modules/spells/tribal_claw.dm]: descript variable names

File: modular_skyrat/master_files/code/modules/spells/tribal_claw.dm [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/obj/effect/proc_holder/spell/aoe_turf/repulse/lizard_tail
+	name = "Tail Sweep"
+	desc = "Throw back attackers with a sweep of your tail."
+	sound = 'sound/magic/tail_swing.ogg'
+	charge_max = 150
+	clothes_req = FALSE
+	antimagic_allowed = TRUE
+	range = 1
+	cooldown_min = 150
```
Comment (Fikou) [inline modular_skyrat/master_files/code/modules/spells/tribal_claw.dm]: seconds defines

File: modular_skyrat/master_files/code/modules/spells/tribal_claw.dm [hunk 5]
```diff
@@ -0,0 +1,24 @@
+/obj/effect/proc_holder/spell/aoe_turf/repulse/lizard_tail
+	name = "Tail Sweep"
+	desc = "Throw back attackers with a sweep of your tail."
+	sound = 'sound/magic/tail_swing.ogg'
+	charge_max = 150
```
Comment (Fikou) [inline modular_skyrat/master_files/code/modules/spells/tribal_claw.dm]: seconds defines

File: modular_skyrat/master_files/code/game/objects/items/granters.dm [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+	desc = "It's completely blank."
+	name = "empty scroll"
+	icon_state = "blankscroll"
+
+/obj/item/book/granter/martial/tribal_claw/already_known(mob/user)
+	if(islizard(user) || isunathi(user))
+		return FALSE
+	else
+		to_chat(user, "<span class='warning'>You try to read the scroll but can't comprehend any of it.</span>")
+		return TRUE
```
Comment (Fikou) [inline modular_skyrat/master_files/code/game/objects/items/granters.dm:25]: this doesnt seem right, are you sure you dont want to return ..() on the lizard stuff?

Comment (Zonespace27) [reply modular_skyrat/master_files/code/game/objects/items/granters.dm:25]: This proc is strange and its parent will return FALSE (which means "yes this is good to go") 100% of the time, so it doesn't matter either way.

**Comment by desertmote** [comment]
open the gate further and allow this for any species with claw and tail.
anthro, vulpkanin, akula, aquatic, etc.
huge no for lizard exclusivity

**Comment by ReynTheLord** [comment]
The only race exclusive item we have, from what I recall, is that one lamp for the moth. Making an entire, relatively strong, martial art which is race locked makes it feel like something which is just for certain people, and not for any antag to enjoy. 

Long story short I don't like this and I will be vocal about it.

**Comment by Zonespace27** [comment]
Thought about this a bit longer, I'm not really happy with how it turned out nor do I think it really fits into a niche that isn't "SCarp but slightly different", might revisit this some time

**Pull request closed without merging**